src/components/PostPicker/index.js

```diff
@@ -0,0 +1,388 @@
+import React, { ReactNode, useCallback, useEffect } from 'react';
+
+import {
+	Button,
+	ButtonGroup,
+	ToolbarButton,
+	TabPanel,
+	CheckboxControl,
+	Flex,
+	FlexItem,
+	Modal,
+	Notice,
+	SearchControl,
+	Spinner,
+} from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import {
+	useState,
+	useMemo,
+} from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+import TermSelector from '../TermSelector';
+
+/**
+ * List of selectable posts
+ *
+ * @param {object} props Component props
+ * @returns {ReactNode} Component
+ */
+function PostList( props ) {
+	const {
+		postType,
+		queryArgs,
+		onChange,
+		values = [],
+		isSortable = false,
+	} = props;
+
+	const queriedPosts = useSelect( ( select ) => {
+		return select( 'core' ).getEntityRecords( 'postType', postType, queryArgs ) ?? [];
```

Comment:
On WP 6.1, we could make use of useEntityRecords (although the project I originally wrote this component for is still on 6.0). But it raises a question, which is should we clarify our supported WP versions for this library? I feel like it's eventually going to be an issue if we're not targeting a clear range.

Reply:
Good thinking. We should agree on this separate to this PR and document it. We could align with Altis support of about a year from Altis release date which would mean 6.0 is supported (Until October) and probably should leave this as-is.